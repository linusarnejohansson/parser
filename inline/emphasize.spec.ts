import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("inline.empasize", () => {
	it("basic", () => {
		const result = parse(new Source(IO.StringReader.create("_emphasize_"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("in text", () => {
		const result = parse(new Source(IO.StringReader.create("This is a text with an _empasize_ in it."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
