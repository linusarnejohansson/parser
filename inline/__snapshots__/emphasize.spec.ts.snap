// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline.empasize basic 1`] = `
Array [
  Object {
    "class": "Inline.Emphasize",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "emphasize",
      },
    ],
  },
]
`;

exports[`inline.empasize in text 1`] = `
Array [
  Object {
    "class": "Inline.Text",
    "value": "This is a text with an ",
  },
  Object {
    "class": "Inline.Emphasize",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "empasize",
      },
    ],
  },
  Object {
    "class": "Inline.Text",
    "value": " in it.",
  },
]
`;
